zerver/actions/create_realm.py

```diff
@@ -167,6 +167,10 @@ def do_create_realm(
         kwargs["org_type"] = org_type
     if enable_spectator_access is not None:
         kwargs["enable_spectator_access"] = enable_spectator_access
+    if is_demo_organization:
+        # Welcome emails are disabled for demo organizations, since
+        # their owners will not have a valid email set.
+        kwargs["send_welcome_emails"] = False
```

Comment:
This is the wrong way to do this; we likely don't want to change the default for sending welcome emails in the organization in general -- just block them for the new owner of this demo organization.

I think probably we should replace this with a change inside `enqueue_welcome_emails` of the form:
```
if user_profile.delivery_email == "":
    # Do not attempt to enqueue welcome emails for users without an email address.
    # The assertions here are to help document the only circumstance under which
    # this condition should be possible.
    assert user_profile.realm.is_demo_organization and realm_creation
    return
```

@laurynmm probably this commit can just be replaced with an independent PR with that revised plan.

Reply:
See also https://github.com/zulip/zulip/pull/19741/commits/f662b0834cb157e4ba161534ee88290a28fa55c3#r932756780 - I think that actually we're going to end up skipping every part of this commit, and just doing the thing described in the last comment.